pkg/authorization/api/validation/validation.go

```diff
@@ -147,7 +147,7 @@ func ValidatePolicy(policy *authorizationapi.Policy, isNamespaced bool) field.Er
 			allErrs = append(allErrs, field.Invalid(keyPath.Child("metadata", "name"), role.Name, "must be "+roleKey))
 		}
 
-		allErrs = append(allErrs, validateRole(role, isNamespaced, keyPath)...)
+		allErrs = append(allErrs, validateRoleMeta(role, isNamespaced, keyPath)...) // only fail policy validation based on roles when it is a metadata error
```

Comment:
If you separate policyupdate from policy create, you'll be able to use the correct validateRole method for each of the cases and still be able to enforce.